src/core/utility/file/filePath/FilePath.cpp

```diff
@@ -72,34 +62,32 @@ bool FilePath::empty() const {
 }
 
 bool FilePath::exists() const noexcept {
-  if(!m_checkedExists) {
-    m_exists = boost::filesystem::exists(getPath());
-    m_checkedExists = true;
+  if(!m_exists.has_value()) {
+    m_exists = fs::exists(getPath());
   }
 
-  return m_exists;
+  return m_exists.value();
```

Comment:
Please invert the if condition

Reply:
m_checkedExists is not existing anymore.
Or you mean what ???